shopelectro/views/catalog.py

```diff
@@ -178,83 +182,27 @@ def merge_products_context(products):
 @set_csrf_cookie
 class CategoryPage(catalog.CategoryPage):
 
-    def get_context_data(self, **kwargs):
-        """Add sorting options and view_types in context."""
-        context = super().get_context_data(**kwargs)
-        products_on_page = int(self.request.GET.get(
-            'step', get_products_count(self.request),
-        ))
-        page_number = int(self.request.GET.get('page', 1))
-        view_type = self.request.session.get('view_type', 'tile')
-        sorting_index = int(self.kwargs.get('sorting', 0))
-        sorting_option = SortingOption(index=sorting_index)
-        category = context['category']
-        if (
-            page_number < 1 or
-            products_on_page not in settings.CATEGORY_STEP_MULTIPLIERS
-        ):
-            raise http.Http404('Page does not exist.')  # Ignore CPDBear
-
-        all_products = models.Product.actives.get_category_descendants(
-            category, ordering=(sorting_option.directed_field, )
+    def get_products(self) -> QuerySet:
+        return models.Product.actives.get_category_descendants(
+            self.object.model
         )
 
-        group_tags_pairs = (
-            models.Tag.objects
-            .filter_by_products(all_products)
-            .get_group_tags_pairs()
+    def get_context_data(self, **kwargs):
+        """Add sorting options and view_types in context."""
+        context_ = (
+            context.Category(self.kwargs, self.request)
+            | context.TaggedCategory()
```

Comment:
Currently I do not understand why we need such syntax: `context_obj | context_obj | ...`.
It has some critical cons:
- We can not explicitly work with dependencies
- We have to note, how some context depends on another one

I propose to use classic object composition, that has no any of these cons

Reply:
e.g. of object composition:
```
category = context.SortedCategory(context.TaggedCategory(context.Category(self.kwargs, self.request)))
pagination = context.PaginationContext(category)
rendered_fields = context.RenderedFields(category)

Contexts(category, pagination, rendered_fields).get_context()
```